dist/Client.js

```diff
@@ -316,6 +356,116 @@ class APIClient {
+   * @param {String} detectorId Id of the detctor involved
+   * @throws {APIError} Containing error code and text
+   */
+
+
+  async addRasterToDetector(rasterId, detectorId) {
+    const response = await this._request(`/detectors/${detectorId}/training_rasters/`, 'POST', {
+      'content-type': 'application/json'
+    }, JSON.stringify({
+      'raster_id': rasterId
+    }));
+    await checkResponse(response);
+  }
+  /**
+   * @async
+   * @function setAnnotations
+   * @summary Overwrites a given annotation type for a raster belonging to a detector training set
+   * @description Creates a detector setting its name and type
+   * @param {String} detectorId If of the detector
+   * @param {String} rasterId Id of the raster
+   * @param {String} annotationType Type of annotation; one of 'outline', 'training_area', 'testing_area', 'validation_area'
+   * @param {Object} annotationsGeoJSon GeoJSON representation of the annotation geometry(ies)"
+   * @throws {APIError} Containing error code and text
+   */
+
+
+  async setAnnotations(detectorId, rasterId, annotationType, annotationsGeoJSon) {
+    let resp, data;
+    const annotationTypes = ['outline', 'training_area', 'testing_area', 'validation_area'];
+    annotationType = annotationType.toLowerCase();
+
+    if (!annotationTypes.includes(annotationType)) {
+      const validTypes = annotationTypes.join(', ');
+      throw new ValidationError(`Invalid annotation type ${annotationType}; allowed values: ${validTypes}.`);
+    } // Get upload URL
+
+
+    resp = await this._request(`/detectors/${detectorId}/training_rasters/${rasterId}/${annotationType}/upload/bulk/`, 'POST');
+    await checkResponse(resp);
+    data = await resp.json();
+    const uploadUrl = data['upload_url'];
+    const uploadId = data['upload_id']; // Put data in remote storage
+
+    resp = await this._request(uploadUrl, 'PUT', {
+      /* 'Content-Length': fileSizeInBytes, */
+      'Content-Type': 'application/json'
+    }, JSON.stringify(annotationsGeoJSon), false);
+    await checkResponse(resp); // Start commit
+
+    resp = await this._request(`/detectors/${detectorId}/training_rasters/${rasterId}/${annotationType}/upload/bulk/${uploadId}/commit/`, 'POST');
+    await checkResponse(resp);
+    data = await resp.json(); // Poll until operation finishes
+
+    await this._waitUntilOperationCompletes(data['operation_id'], data['poll_interval']);
+  }
+  /**
+   * @async
+   * @function trainDetector
+   * @summary Trains a detector
+   * @description Launches a training on a detector and waits until it finishes
+   * @param {String} detectorId Identifier for the detector
+   * @throws {APIError} Containing error code and text
+   */
+
+
+  async trainDetector(detectorId) {
+    // Launch training
+    const response = await this._request(`/detectors/${detectorId}/train/`, 'POST');
+    await checkResponse(response);
+    const data = await response.json(); // Poll until training finishes
```

Comment:
comment should be on next line ? But generally, I feel like this doesn't add much value given the next line is doing a 'wait', should be pretty obvious we're waiting for the operation to finish

Reply:
Checked